src/server/ProcessList.cpp

```diff
@@ -31,23 +31,20 @@ using namespace Microsoft::Console::Interactivity;
 {
     assert(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked());
 
-    auto pProcessData = FindProcessInList(dwProcessId);
-    if (pProcessData)
+    if (FindProcessInList(dwProcessId))
     {
-        return E_FAIL;
+        return S_FALSE;
     }
 
+    std::unique_ptr<ConsoleProcessHandle> pProcessData;
     try
     {
-        pProcessData = new ConsoleProcessHandle{ dwProcessId, dwThreadId, ulProcessGroupId };
-        _processes.emplace_back(pProcessData);
+        pProcessData = std::make_unique<ConsoleProcessHandle>(dwProcessId, dwThreadId, ulProcessGroupId);
+        _processes.emplace_back(pProcessData.get());
     }
     CATCH_RETURN();
 
-    if (ppProcessData)
-    {
-        *ppProcessData = pProcessData;
-    }
+    wil::assign_to_opt_param(ppProcessData, pProcessData.release());

```

Comment:
SCARY! This will delete the processdata if nobody asks for the optional output param

Reply:
wait, no it won't. that's even scarier lol